xl-v1-object.go

```diff
@@ -429,6 +452,7 @@ func (xl xlObjects) PutObject(bucket string, object string, size int64, data io.
 		if vErr := data.(*signVerifyReader).Verify(); vErr != nil {
 			// Incoming payload wrong, delete the temporary object.
 			xl.deleteObject(minioMetaTmpBucket, tempObj)
+
 			// Error return.
```

Comment:
before returning cache used size must be reduced `c.currentSize -= uint64(size)` should be done, by doing maybe newBuffer.CloseWIthError(err)
better not do caching for put-object for 2 reasons:
1) we can keep the cache code simple by not handling special case of sha/signature mismatch.
2) we will never cache something that will never be read.